### PR TITLE
Ensure that jwt iss/username combination is unique 

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Follow up to https://github.com/crate/crate/commit/4cfd32e3e79632d55bb6e2d67a78e0b1510cb767

### CODE REVIEW

File: server/src/main/java/io/crate/role/RoleManagerService.java [hunk 1]
```diff
@@ -116,7 +116,11 @@ public CompletableFuture<Long> createRole(String roleName,
                                               @Nullable JwtProperties jwtProperties) {
         return transportCreateRoleAction.execute(new CreateRoleRequest(roleName, isUser, hashedPw, jwtProperties), r -> {
             if (r.doesUserExist()) {
-                throw new RoleAlreadyExistsException(roleName);
+                if (jwtProperties == null) {
+                    throw new RoleAlreadyExistsException(roleName);
+                } else {
+                    throw new RoleAlreadyExistsException(roleName, jwtProperties);
+                }
             }
```
Comment (BaurzhanSakhariev) [inline server/src/main/java/io/crate/role/RoleManagerService.java:124]: Another approach could be introducing `JwtPropertiesAlreadyExists` exception and adding jwtExists to `WriteRoleResponse`.

I felt like having user with conflicting jwt props kinda fits into "role exists" but not sure. wdyt?

File: docs/sql/statements/create-user.rst [hunk 2]
```diff
@@ -79,3 +79,14 @@ The following ``user_parameter`` are supported to define a new user account:
...
+:jwt:
+  JWT properties map ('iss' and 'username') entered as string literal. e.g.::
+
+     CREATE USER john WITH (jwt = {"iss" = 'https://login.company.org/keys', "username" = 'test@company.org'})
+
+  `iss`_ is a JWK endpoint, containing public keys.
+  ``username`` is a user name in a third party app. Not neccearily have to match CrateDb user.
+  Combination of ``iss`` and ``username`` must be unique.
+
+.. _iss: https://www.rfc-editor.org/rfc/rfc7519#section-4.1.1
```
Comment (BaurzhanSakhariev) [inline docs/sql/statements/create-user.rst:92]: Once signature verification with docs is merged, I will also add here "see also" (ref to jwt_auth docs) link.

File: server/src/test/java/io/crate/integrationtests/RoleManagementIntegrationTest.java [hunk 3]
```diff
@@ -277,4 +277,14 @@ public void testDropSuperUserThrowsException() {
...
+
+    @Test
+    public void test_create_user_jwt_properties_must_be_unique() {
+        execute("CREATE USER user1 WITH (jwt = {\"iss\" = 'dummy.org/keys', \"username\" = 'app_user'})");
+        assertUserIsCreated("user1");
+        Asserts.assertSQLError(() -> execute("CREATE USER user2 WITH (jwt = {\"iss\" = 'dummy.org/keys', \"username\" = 'app_user'})"))
+            .hasPGError(INTERNAL_ERROR)
+            .hasHTTPError(CONFLICT, 4099)
+            .hasMessageContaining("Role 'user2' or another role with the same combination of jwt properties already exists");
+    }
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/io/crate/integrationtests/RoleManagementIntegrationTest.java:289]: I will remove that, just added to find out a working combination of literals to enter map.

Was not obvious, spent some time find working combination of literals, that's why added an example)

Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/integrationtests/RoleManagementIntegrationTest.java:289]: Update: actually, given that we need to query sys.users to test new exposed property, maybe it an stay.

File: server/src/test/java/io/crate/role/TransportRoleActionTest.java [hunk 4]
```diff
@@ -60,6 +60,23 @@ public void testCreateFirstUser() throws Exception {
         assertThat(metadata.roles().get("root").jwtProperties().username()).isEqualTo("test");
     }
 
+    @Test
+    public void test_create_user_with_mathcing_jwt_props_exists() throws Exception {
```
Comment (seut) [inline server/src/test/java/io/crate/role/TransportRoleActionTest.java:64]:
```suggestion
    public void test_create_user_with_matching_jwt_props_exists() throws Exception {
```

File: server/src/main/java/io/crate/exceptions/RoleAlreadyExistsException.java [hunk 5]
```diff
@@ -23,9 +23,17 @@
...
+
 public class RoleAlreadyExistsException extends RuntimeException implements ConflictException, UnscopedException {
 
     public RoleAlreadyExistsException(String roleName) {
         super(String.format(Locale.ENGLISH, "Role '%s' already exists", roleName));
     }
+
+    public RoleAlreadyExistsException(String roleName, @NotNull JwtProperties jwtProperties) {
+        super(String.format(Locale.ENGLISH, "Role '%s' or another role with the same combination of jwt properties already exists", roleName));
+    }
```
Comment (seut) [inline server/src/main/java/io/crate/exceptions/RoleAlreadyExistsException.java:38]: Having `jwtProperties` unused to use a different ctor looks not great.
Maybe use a factory method and add a default private ctor?
```suggestion
    public RoleAlreadyExistsException for(String roleName, @Nullable JwtProperties jwtProperties) {
        if (jwtProperties == null) {
            return new RoleAlreadyExistsException(String.format(Locale.ENGLISH, "Role '%s' already exists", roleName));
        } else {
            return new RoleAlreadyExistsException(String.format(Locale.ENGLISH, "Role '%s' or another role with the same combination of jwt properties already exists", roleName));
        }
    }
```

this way no `jwtProperties` null check must be done outside.

Comment (BaurzhanSakhariev) [reply server/src/main/java/io/crate/exceptions/RoleAlreadyExistsException.java]: Thanks, applied and adapted depending code

**Pull request merged**